documentation/docs/scenarios/changeManagement.md

```diff
@@ -0,0 +1,343 @@
+```
+@Library('piper-library-os@master') _
+
+// the shared lib containing the additional configuration
+// must be configured in Jenkins
+@Library(foo@master') __
+
+// in the shared lib denoted by 'foo', the additional configuration file
+// must be located under 'resources' ('resoures/myConfig.yml')
+prepareDefaultValues script: this, customDefaults: 'myConfig.yml'
+```
+Example content of `resources/myConfig.yaml` in branch `master` of the repository denoted by `foo`:
+```
+general:
+  changeManagement:
+    changeDocumentLabel: 'ChangeDocument\s?:'
+    cmClientOpts: '-Djavax.net.ssl.trustStore=<path to truststore>'
+    credentialsId: 'CM'
+    type: 'SOLMAN'
+    endpoint: 'https://example.org/cm'
+    git:
+      from: 'HEAD~1'
+      to: 'HEAD'
+      format: '%b'
+```
+The properties configured in section `general/changeManagement` are shared between all steps related to change management.
+You can also configure the properties on a per-step basis, for example:
+```
+[...]
+ steps:
+   transportRequestCreate:
+     changeManagement:
+       type: 'SOLMAN'
+       endpoint: 'https://example.org/cm'
+       [...]
+```
+The parameters can also be provided when the step is invoked. See [Examples](#Examples).
+
+### Result
+
+As a result, you get the ID of the newly created transport request.
+
+For exceptions, see **Exceptions** in [transportRequestCreate](https://sap.github.io/jenkins-library/steps/transportRequestCreate/).
+
+### Example
+
+```
+def transportRequestId = transportRequestCreate script:this,
+                                                changeDocumentId: '001,'
+                                                changeManagement: [
+                                                  type: 'SOLMAN'
+                                                  endpoint: 'https://example.org/cm'
+                                                ]
+```
+
+## Upload a File to a Transport Request (`transportRequestUploadFile`)
+
+Upload a file to your transport request for a change document in SAP Solution Manager.
+
+### Prerequisites
+
+You have built your Java sources. For an example, see [mtaBuild](https://sap.github.io/jenkins-library/steps/mtaBuild/).
+
+### Mandatory Parameters
+
+| Parameter | Description |
+| --- | --- |
+| `script` | The common script envoronment of the running Jenkinsfile. The reference to the script that calls the pipeline step is privided by the `this` parameter, as in `script: this`. This allows the function to access the `commonPipelineEnvironment` to retrieve configuration parameters. |
+| `changeDocumentId` | The ID of the change document related to the transport request that is to be released. The ID is retrieved from the Git commit history. |
+| `transportRequestId` | The ID of the transport request to release. The ID is retreved from the Git commit history. |
+| `applicationId` | The ID of the application. |
```

Comment:
ApplicationId is to my best knowledge only used in case of SOLMAN. SOLMAN uses it for decission making what should happen with the binary. Maybe we get a closer explanation from some solman colleagues ...